dart/dynamics/Joint.h

```diff
@@ -119,6 +126,14 @@ class Joint
   /// Get number of generalized coordinates
   DEPRECATED(4.1) virtual size_t getDof() const = 0;
 
+  /// Get an object to access the _index-th degree of freedom (generalized
+  /// coordinate) of this Joint
+  virtual DegreeOfFreedom* getDof(size_t _index) = 0;
+
+  /// Get an object to access the _index-th degree of freedom (generalized
+  /// coordinate) of this Joint
+  virtual const DegreeOfFreedom* getDof(size_t _index) const = 0;
```

Comment:
I see what you're saying. It's true that the DOF belongs to its joint, but one of the purposes of the DOF class is to offer a more flexible way of accessing joint properties. In particular, I think it's valuable for the user to be able to customize DOF names however they'd like to, and then access those names through the Skeleton::getDof(string) function. For that to work, we need the Skeleton to manage DOF name uniqueness. We could enforce name uniqueness requiring a [JointName].[DofName] style as you're suggesting, but that would detract from customizability.